syntax = "proto3";
package seismic_core;

// The seismic core service definition.
service Core {
  rpc ShatterLink (ShatterLinkRequest) returns (ShatterReply) {}
  rpc StitchSurface (SurfaceRequest) returns (SurfaceReply) {}
}

message  SurfaceRequest {
    string surface    = 1;
    string basename   = 2;
    int32  cubexs     = 3;
    int32  cubeys     = 4;
    int32  cubezs     = 5;
    int32  fragmentxs = 6;
    int32  fragmentys = 7;
    int32  fragmentzs = 8;
}

message SurfaceValue {
    int64  i = 1;
    double v = 2;
}

message  SurfaceReply {
  repeated SurfaceValue surface = 1;
}

message ShatterLinkRequest {
  string link = 1;
}

message ShatterReply {
  string link = 1;
}
